### PR TITLE
Default to OpenSSL 1.0

### PR DESCRIPTION
**Pull request by healther**

### CODE REVIEW

File: var/spack/repos/builtin/packages/openssl/package.py [hunk 1]
```diff
@@ -33,17 +33,14 @@ class Openssl(Package):
        Layer Security (TLS) and Secure Sockets Layer (SSL) protocols.
        It is also a general-purpose cryptography library."""
     homepage = "http://www.openssl.org"
-    url      = "http://www.openssl.org/source/openssl-1.0.1h.tar.gz"
+    url = "https://www.openssl.org/source/openssl-1.0.1h.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openssl/package.py]: Can you remove this change to URL? We just finished a PR that bounced back and forth between versions and we decided on http.

**Comment by citibeth** [comment]
Adding preferred=True requires more discussion.  If you would like to use `1.0.2k` for your builds, you can set that in your `packages.yaml` file.

**Comment by healther** [comment]
The reason for the preferred setting was due to the lack of support on part of some packages. I'd say that it is helpful, but if you prefer not to use it, we can close this PR.

**Comment by citibeth** [comment]
> The reason for the preferred setting was due to the lack of support on part of some packages. I'd say that it is helpful, but if you prefer not to use it, we can close this PR.

I see.  That is indicated in the long note.  I'm OK with preferred=True, but please do restore the original Note explaining it.

**Comment by citibeth** [comment]
OK, I can approve as soon as the URL is changed back to `http://`.  See #3113

File: var/spack/repos/builtin/packages/openssl/package.py [hunk 2]
```diff
@@ -33,17 +33,19 @@ class Openssl(Package):
        Layer Security (TLS) and Secure Sockets Layer (SSL) protocols.
        It is also a general-purpose cryptography library."""
     homepage = "http://www.openssl.org"
-    url      = "http://www.openssl.org/source/openssl-1.0.1h.tar.gz"
-    list_url = "https://www.openssl.org/source/old/"
+    url = "http://www.openssl.org/source/openssl-1.0.1h.tar.gz"
+    list_url = "http://www.openssl.org/source/old/"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/openssl/package.py]: With `http`, `spack versions` and `spack checksum` no longer work. Is there a reason you made this change?

**Comment by citibeth** [comment]
I'm find with whatever is needed on `list_url`, as long as `url` remains
`http://`.

File: var/spack/repos/builtin/packages/openssl/package.py (hunk 2 above)
Comment (healther) [inline var/spack/repos/builtin/packages/openssl/package.py]: That was for consistency (I hadn't had the time to check on your reasoning on http > https and assumed that it was wanted everywhere), I'll change this back and then read up

**Comment by healther** [comment]
@adamjstewart your change request is corrected right?

**Comment by healther** [comment]
Is there still something to do for me?

**Comment by adamjstewart** [comment]
You might want to get rid of the comment that says "We could thus make this version the preferred version, if we find that many packages cannot handle version 1.1."

**Pull request merged**